### PR TITLE
Further shorten the addition chain for scalar inversion.

### PR DESCRIPTION
**Pull request by briansmith**
Reduce the number of squarings by one and reduce the number of
multiplications by three.

### CODE REVIEW

File: src/scalar_impl.h [hunk 1]
```diff
@@ -69,18 +69,21 @@ static void secp256k1_scalar_inverse(secp256k1_scalar *r, const secp256k1_scalar
...
     secp256k1_scalar_mul(&u5, &u2, &x2);
     secp256k1_scalar_mul(&x3, &u5,  &u2);
+    secp256k1_scalar_mul(&u9, &x3, &u2);
+    secp256k1_scalar_mul(&u11, &u9, &u2);
+    secp256k1_scalar_mul(&u13, &u11, &u2);
 
-    secp256k1_scalar_sqr(&x6, &x3);
-    for (i = 0; i < 2; i++) {
+    secp256k1_scalar_sqr(&x6, &u13);
+    for (i = 0; i < 1; i++) {
```
Comment (peterdettman) [inline src/scalar_impl.h]: @briansmith Could you squash this redundant loop before I merge please?

Comment (briansmith) [reply src/scalar_impl.h]: Done.

**Comment by sipa** [comment]
ACK for merging in bitcoin-core/secp256k1, see https://github.com/bitcoin-core/secp256k1/pull/452#issuecomment-297178903

**Comment by peterdettman** [comment]
Merged.

**Pull request merged**